src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -692,77 +694,46 @@ public Node call() throws Exception {
                 Pod pod = getPodTemplate(slave, unwrappedTemplate);
 
                 String podId = pod.getMetadata().getName();
-                String namespace = Strings.isNullOrEmpty(t.getNamespace())
-                        ? client.getNamespace()
-                        : t.getNamespace();
+                String namespace = Strings.isNullOrEmpty(t.getNamespace()) ? client.getNamespace() : t.getNamespace();
 
-                LOGGER.log(Level.FINE, "Creating Pod: {0} in namespace {1}", new Object[] { podId, namespace });
+                String[] logArgs = new String[] { namespace, podId };
+                LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", logArgs);
                 pod = client.pods().inNamespace(namespace).create(pod);
-                LOGGER.log(Level.INFO, "Created Pod: {0} in namespace {1}", new Object[] { podId, namespace });
 
-                // We need the pod to be running and connected before returning
-                // otherwise this method keeps being called multiple times
-                ImmutableList<String> validStates = ImmutableList.of("Running");
```

Comment:
This was the only instance of `ImmutableList` in use, you can remove the import.